### PR TITLE
Implement afterEach mode

### PR DESCRIPTION
**Pull request by papandreou**
Also allow queueing up HTTP traffic by calling fetchception multiple times.

Copy/paste job from httpception.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -124,19 +156,29 @@ function fetchception(expectedExchanges, promiseFactory) {
...
+            )
+            .finally(() => restoreFetch());
+    } else {
+        promiseForAfterEach = expectWithoutFootgunProtection(function () {
+            return expect.promise((resolve, reject) => {
+                resolveNext = resolve;
+            });
+        }, 'not to error').then(
+            () => verifyConversation(mockDefinitionForTheCurrentTest, httpConversation)
+        );
```
Comment (gustavnikolaj) [inline index.js]: `restoreFetch` is not being called in this case, is it?

**Comment by papandreou** [comment]
You're right that the failure cases of the `afterEach` mode are not tested at all. I'll try to come up with something :)

File: index.js (hunk 1 above)
Comment (papandreou) [inline index.js]: You're right! Fixed.

**Comment by papandreou** [comment]
Added some tests... And of course it exposed a rather serious bug (also fixed :)

**Comment by papandreou** [comment]
Released in 0.1.0.

**Pull request merged**